### PR TITLE
Fix schema generation using interface relationships and subscriptions

### LINKED ISSUE #3439: Error when using the subscription plugin in combination with interfaces and relationship directive
**Background**
I want to write a custom subscription plugin that allows me to log which nodes have been created or updated in the database (see https://github.com/neo4j/graphql/issues/1913). I have already written a simple custom plugin. However, so far I could not test it yet, because already at the start of the application an error occur in combination with our schema (the schema below is not part of our current schema, I just reproduced the problem with it).

**Describe the bugs**
In the schema it happens that types have to implement several interfaces. For some reason the "inheritance" of the relationship directive does not work properly when a type implements multiple interfaces. Therefore we currently cannot use the relationship directive on the property in the interface, instead the directive is set to the properties of the implementations (see below)

However, when starting the application with subscription plugin enabled, I then get an error message.

**Type definitions**
````graphql
interface INode {
  id: String!
}

interface IProduct implements INode {
  id: String!

  name: String!
  genre: Genre!
}

type Movie implements INode & IProduct {
  id: String!

  name: String!
  genre: Genre! @relationship(type: "HAS_GENRE", direction: OUT)
}

type Series implements INode & IProduct {
  id: String!

  name: String!
  genre: Genre! @relationship(type: "HAS_GENRE", direction: OUT)
}

type Genre {
  name: String! @unique
  product: [IProduct!]! @relationship(type: "HAS_GENRE", direction: IN)
}
````

**To Reproduce**

1. Run a server with the following code...
````typescript
...
const neo4jGraphQLConstructor: Neo4jGraphQLConstructor = {
  typeDefs,
  driver,
  plugins: {
    subscriptions: new Neo4jGraphQlSubscriptionActivityLogPlugin(),
  }
}
...
````
2. See error
````
Error: Interface field IProductEventPayload.genre expected but MovieEventPayload does not provide it.

Interface field IProductEventPayload.genre expected but SeriesEventPayload does not provide it.
````

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

Fix schema generation using interface relationships and subscriptions.

We were adding relationship fields into interface event payloads for nested relationships, which subscriptions does not support.

Some changes to an unrelated schema test because that test contains types which only contain relationships.

## Complexity

Complexity: Low

# Issue

Closes #3439

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3678.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 32 | 37 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 72 | 86 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 50 | 52 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3035 | 3151 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2887 | 2873 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 39 | 57 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 40 | 54 | 2174 |
|  aggregations.DeeplyNestedCount | 13070333 | 13070331 | 4544 | 4180 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 134 | 155 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 83 | 71 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 187 | 198 | 3003 |
|  connections.Connection | 14082 | 14082 | 77 | 68 | 2174 |
|  connections.NestedConnection | 45472 | 45470 | 137 | 130 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 64 | 68 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 186 | 328 | 1040 |
|  create.SimpleMutation | 7 | 7 | 52 | 62 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 615 | 698 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153617 | 153617 | 176 | 190 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 233 | 241 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 153 | 191 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 68 | 44 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 94 | 97 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 40 | 41 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 57 | 62 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 25 | 28 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 45 | 56 | 2174 |
|  query.QueryWhere | 9695 | 9707 | 32 | 51 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9873 | 9885 | 47 | 57 | 2161 |
|  query.Nested | 10096041 | 10096041 | 10017 | 10559 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10036 | 10274 | 2004000 |
|  query.OrFilterOnRelationships | 42056 | 43083 | 203 | 202 | 1642 |
|  query.OrFilterOnRelationshipsAndNested | 35945 | 36548 | 211 | 256 | 1642 |
|  query.QueryWithNestedIn | 13981 | 15145 | 53 | 65 | 1409 |
|  query.NestedConnectionWhere | 9834 | 9834 | 54 | 63 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10019 | 10069 | 71 | 89 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19212 | 19302 | 131 | 148 | 1624 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 300 | 225 | 1134 |
|  query.Fulltext | 80 | 80 | 38 | 32 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 51 | 60 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 63 | 91 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1505 | 1475 | 157 | 288 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 53 | 53 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 93 | 95 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 128 | 173 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 67 | 64 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 63 | 62 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 289 | 313 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 342 | 452 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 95 | 128 | 2002 |

</details>
        
Old Schema Generation: 38.417s
Schema Generation: 38.880s
Old Subgraph Schema Generation: 47.394s
Subgraph Schema Generation: 49.415s

File: packages/graphql/tests/schema/issues/3439.test.ts [hunk 1]
```diff
@@ -0,0 +1,3214 @@
...
+
+import { printSchemaWithDirectives } from "@graphql-tools/utils";
+import { gql } from "graphql-tag";
+import { lexicographicSortSchema } from "graphql/utilities";
+import { Neo4jGraphQL } from "../../../src";
+import { TestSubscriptionsPlugin } from "../../utils/TestSubscriptionPlugin";
+import { validateSchema } from "graphql";
+
+describe("https://github.com/neo4j/graphql/issues/3439", () => {
+    test("Example 1", async () => {
```
Comment (angrykoala) [inline packages/graphql/tests/schema/issues/3439.test.ts:28]: A bit more descriptive title could be good :)

File: packages/graphql/tests/schema/issues/3439.test.ts [hunk 2]
```diff
@@ -0,0 +1,3214 @@
...
+ */
+
+import { printSchemaWithDirectives } from "@graphql-tools/utils";
+import { gql } from "graphql-tag";
+import { lexicographicSortSchema } from "graphql/utilities";
+import { Neo4jGraphQL } from "../../../src";
+import { TestSubscriptionsPlugin } from "../../utils/TestSubscriptionPlugin";
+import { validateSchema } from "graphql";
+
+describe("https://github.com/neo4j/graphql/issues/3439", () => {
```
Comment (angrykoala) [inline packages/graphql/tests/schema/issues/3439.test.ts:27]: Maybe we could add an integration/e2e test to replicate this issue?

Schema tests are good but a bit unreliable on their own

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/tests/schema/issues/3439.test.ts (hunk 2 above)
Comment (darrellwarde) [reply packages/graphql/tests/schema/issues/3439.test.ts:27]: The magic sauce of this schema test is that I added validation using `validateSchema` - this is an exact reproduction of this issue in that they weren't able to generate the schema, and these tests demonstrate that they now can.

**Pull request merged**